src/compiler/transformers/esnext.ts

```diff
@@ -168,48 +162,58 @@ namespace ts {
             }
 
             const target = createConditional(
-                createLogicalOr(
-                    createStrictEquality(createAssignment(temp, leftExpression), createNull()),
-                    createStrictEquality(temp, createVoidZero())
-                ),
+                createNotNullCondition(capturedLeft),
+                rightExpression,
                 createVoidZero(),
-                rightExpression
             );
             return thisArg ? createSyntheticReferenceExpression(target, thisArg) : target;
         }
 
-        function createNotNullCondition(node: Expression) {
+        function createNotNullCondition(expression: CapturedExpression) {

```

Comment:
If you add an `invert` argument here, you could invert the condition and swap the branches:

```ts
createToken(invert ? SyntaxKind.EqualsEqualsEqualsToken : SyntaxKind.ExclamationEqualsEqualsToken)
createToken(invert ? SyntaxKind.BarParToken : SyntaxKind.AmpersandAmpersandToken)
```

Reply:
done